### PR TITLE
Add support for Redis for simplified init

### LINKED ISSUE #2902: Add Redis as a DB option in simplified init
Start with managed add-on Redis for Day 0.

### PR DESCRIPTION
**Pull request by weikanglim**
Add support for Redis as an in-memory DB.

Python and JS libraries are autodetected. The bicep generated will host Redis using Azure Container Apps' managed add-on.

Tests have been added to validate bicep correctness which largely covers the feature. Manual tests were ran additionally to verify the Container App functionality.

Fixes #2902

### CODE REVIEW

**Review by wbreza — COMMENTED** [review]
Is there a way we can data drive some of this detection from an internal configuration file vs having coded flows for each new type?

**Comment by weikanglim** [comment]
@wbreza Certainly possible, but if we ever needed to support anything more than a simple string-key match, we end up needing more complicated expressions. Not sure what we gain by extracting out matching into a declarative configuration file since it doesn't need to be user-defined currently.

File: cli/azd/internal/repository/detect_confirm.go [hunk 1]
```diff
@@ -175,6 +183,8 @@ func (d *detectConfirm) render(ctx context.Context) error {
 			recommendedServices = append(recommendedServices, "Azure Database for PostgreSQL flexible server")
 		case appdetect.DbMongo:
 			recommendedServices = append(recommendedServices, "Azure CosmosDB API for MongoDB")
+		case appdetect.DbRedis:
```
Comment (vhvb1989) [inline cli/azd/internal/repository/detect_confirm.go:186]: Can we have tests for app-detect, calling `DetectProject()` from all the supported projects to add coverage and protect againts regressions?

Comment (weikanglim) [reply cli/azd/internal/repository/detect_confirm.go:186]: We do have these tests, I can update

Comment (weikanglim) [reply cli/azd/internal/repository/detect_confirm.go:186]: done

**Review by vhvb1989 — APPROVED** [review]
Thank you for adding the tests! :)

**Pull request merged**